CAMDU/Beta/Intensity_Projection.py

```diff
@@ -0,0 +1,206 @@
+import omero
+import omero.scripts as scripts
+from omero.gateway import BlitzGateway, DatasetWrapper
+from omero.rtypes import rlong, rstring, robject
+import omero.util.script_utils as script_utils
+import numpy as np
+'''
+Slow, but low memory usage
+'''
+
+
+def log(data):
+    """Handle logging or printing in one place."""
+    print(data)
+
+
+def copyMetadata(conn, newImage, image):
+    """
+    Copy important metadata
+    Reload to prevent update conflicts
+    """
+    newImage = conn.getObject("Image", newImage.getId())
+    new_pixs = newImage.getPrimaryPixels()._obj
+    old_pixs = image.getPrimaryPixels()._obj
+    new_pixs.setPhysicalSizeX(old_pixs.getPhysicalSizeX())
+    new_pixs.setPhysicalSizeY(old_pixs.getPhysicalSizeY())
+    new_pixs.setPhysicalSizeZ(old_pixs.getPhysicalSizeZ())
+    conn.getUpdateService().saveObject(new_pixs)
+    for old_channels, new_channels in zip(image.getChannels(),
+                                          newImage.getChannels()):
+        new_LogicChan = new_channels._obj.getLogicalChannel()
+        new_LogicChan.setName(rstring(old_channels.getLabel()))
+        new_LogicChan.setEmissionWave(old_channels.getEmissionWave(units=True))
+        new_LogicChan.setExcitationWave(
+            old_channels.getExcitationWave(units=True))
+        conn.getUpdateService().saveObject(new_LogicChan)
+
+    if newImage._prepareRenderingEngine():
+        newImage._re.resetDefaultSettings(True)
+
+
+def getImages(conn, script_params):
+    """
+    Get the images
+    """
+    message = ""
+    objects, log_message = script_utils.get_objects(conn, script_params)
+    message += log_message
+    if not objects:
+        return None, message
+
+    data_type = script_params["Data_Type"]
+
+    if data_type == 'Dataset':
+        images = []
+        for ds in objects:
+            images.extend(list(ds.listChildren()))
+        if not images:
+            message += "No image found in dataset(s)"
+            return None, message
+    else:
+        images = objects
+    return images
+
+
+def getRoiShape(s):
+    shape = {}
+    shape['x'] = int(np.floor(s.getX().getValue()))
+    shape['y'] = int(np.floor(s.getY().getValue()))
+    shape['w'] = int(np.floor(s.getWidth().getValue()))
+    shape['h'] = int(np.floor(s.getHeight().getValue()))
+    return shape
+
+
+def planeGenerator(new_Z, C, T, Z, pixels, projection, shape=None):
+    """
+    Set up generator of 2D numpy arrays, each of which is a MIP
+    To be passed to createImage method so must be order z, c, t
+    """
+    for z in range(new_Z):  # createImageFromNumpySeq expects Z, C, T order
+        for c in range(C):
+            for t in range(T):
+                for eachz in range(Z[0]-1, Z[1]):
+                    plane = pixels.getPlane(eachz, c, t)
+                    if shape is not None:
+                        plane = plane[shape['y']:shape['y']+shape['h'],
+                                      shape['x']:shape['x']+shape['w']]
```

Comment:
You could use `plane = pixels.getTile(Z, C, T (x, y, w, h))` here, to avoid loading the whole plane
https://github.com/ome/omero-py/blob/2b9d1b43f69e02d8784bfef50dd07415eb193f34/src/omero/gateway/__init__.py#L7561